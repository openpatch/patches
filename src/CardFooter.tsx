/** @jsx jsx */
import { ReactNode, forwardRef, Ref } from "react";
import { jsx, css } from "@emotion/core";
import { Theme } from "./themes/types";
import { Box } from "./Box";

export type CardFooterProps = {
  noSpacing?: boolean;
  children?: ReactNode;
  align?: "left" | "center" | "right";
};

export const CardFooter = ({
  children,
  noSpacing,
  align = "right",
}: CardFooterProps) => {
  return (
    <Box
      display="flex"
      alignItems="center"
      backgroundColor="neutral.50"
      p="standard"
      css={(theme: Theme) => [
        css`
          justify-content: ${align === "right" && `flex-end`};
          justify-content: ${align === "left" && `flex-start`};
          justify-content: ${align === "center" && `center`};
        `,
        !noSpacing &&
          css`
            & > :not(:first-of-type) {
              margin-left: ${theme.space["standard"]};
            }
          `,
      ]}
    >
      {children}
    </Box>
  );
};
