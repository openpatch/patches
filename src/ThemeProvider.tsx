import React, { ReactElement } from "react";
import { ThemeProvider as EThemeProvider } from "emotion-theming";
import { Theme, base } from "./themes";
import { GlobalCssTheme } from "./themes/utils";

interface ThemeProviderProps {
  theme?: Theme;
  children: ReactElement;
}

const ThemeProvider = ({ theme = base, children }: ThemeProviderProps) => {
  return (
    <EThemeProvider theme={theme}>
      <GlobalCssTheme theme={theme} />
      {children}
    </EThemeProvider>
  );
};

export default ThemeProvider;
