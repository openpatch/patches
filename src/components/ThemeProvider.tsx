import React, { ReactElement } from "react";
import {
  ThemeProvider as SCThemeProvider,
  DefaultTheme,
} from "styled-components";
import defaultTheme from "../shared/theme";

interface ThemeProviderProps {
  theme?: DefaultTheme;
  children: ReactElement;
}

const ThemeProvider = ({
  theme = defaultTheme,
  children,
}: ThemeProviderProps) => {
  return <SCThemeProvider theme={theme}>{children}</SCThemeProvider>;
};

export default ThemeProvider;
