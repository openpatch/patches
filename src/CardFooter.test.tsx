import React from "react";
import { render } from "@testing-library/react";
import { CardFooter, CardFooterProps } from "./CardFooter";
import { ThemeProvider } from "./ThemeProvider";

const renderWithTheme = (props?: CardFooterProps) =>
  render(
    <ThemeProvider>
      <CardFooter {...props} />
    </ThemeProvider>
  );

test("should render", () => {
  renderWithTheme();
});

test("should match snapshot", () => {
  const { container } = renderWithTheme();
  expect(container).toMatchSnapshot();
});
