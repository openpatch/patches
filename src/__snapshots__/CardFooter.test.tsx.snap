// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
.emotion-0 {
  box-sizing: border-box;
  min-width: 0;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #F5F7FA;
  padding: 16px;
  -webkit-box-pack: end;
  -ms-flex-pack: end;
  -webkit-justify-content: flex-end;
  justify-content: flex-end;
}

.emotion-0>:not(:first-of-type) {
  margin-left: 16px;
}

<div>
  <div
    id="patches-root"
  >
    <div
      class="emotion-0"
      display="flex"
    />
  </div>
</div>
`;
